From 781dee40c6f32941207322561ebab7166a90569c Mon Sep 17 00:00:00 2001
From: Martin Briza <mbriza@redhat.com>
Date: Fri, 23 Jun 2017 16:30:09 +0200
Subject: [PATCH] qt: Prefer XCB over Wayland

Wayland users now have no choice (except setting WAYLAND_DISPLAY to '') how to avoid getting VLC under Wayland which may not be optimal yet.

This patch ensures the XCB backend is preferred. This means the developers who want to work on Wayland support will have to clear their DISPLAY variable instead.

Related to https://trac.videolan.org/vlc/ticket/18383 .
---
 modules/gui/qt/qt.cpp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/modules/gui/qt/qt.cpp b/modules/gui/qt/qt.cpp
index 2ad68eb401..3224361ecb 100644
--- a/modules/gui/qt/qt.cpp
+++ b/modules/gui/qt/qt.cpp
@@ -411,16 +411,16 @@ static int Open( vlc_object_t *p_this, bool isDialogProvider )
     intf_thread_t *p_intf = (intf_thread_t *)p_this;
     void *(*thread)(void *) = Thread;
 
-#ifdef QT5_HAS_WAYLAND
-    if( HasWayland() )
-        thread = ThreadWayland;
-    else
-#endif
 #ifdef QT5_HAS_X11
     if( HasX11( p_this ) )
         thread = ThreadXCB;
     else
 #endif
+#ifdef QT5_HAS_WAYLAND
+    if( HasWayland() )
+        thread = ThreadWayland;
+    else
+#endif
 #if defined (QT5_HAS_X11) || defined (QT5_HAS_WAYLAND)
         return VLC_EGENERIC;
 #endif
-- 
2.13.0

